.github/workflows/check-links.yaml

```diff
@@ -0,0 +1,42 @@
+name: docs
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  link-checker:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+      - uses: actions/checkout@v2
+      
+      # Using markdown-link-check to check markdown docs
+      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+        with:
+          use-verbose-mode: 'yes'
+          config-file: '.github/workflows/markdown-link-check-config.json'
+          max-depth: 1
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r userdocs/requirements.txt
+      - name: Build docs for link check
+        run: cd userdocs; mkdocs build
+        # Using liche action to check generated HTML site
+      - name: Link Checker (generated site)
+        id: lc
+        uses: peter-evans/link-checker@v1
+        with:
+          args: -d userdocs/site -r userdocs/site -v
+      - name: Fail if there were link errors
+        run: exit ${{ steps.lc.outputs.exit_code }}
```

Comment:
do we need to do the same here for markdown checker ? or md check will just fail and stop right away ?

Reply:
Yes, it will stop right away. Not sure why they are implemented differently.